deku-derive/src/macros/deku_read.rs

```diff
@@ -352,7 +362,36 @@ fn emit_enum(input: &DekuData) -> Result<TokenStream, syn::Error> {
         });
     }
 
-    // println!("{}", tokens.to_string());
+    // create `deku_id` function for retrieving at runtime the id value
+    let deku_id_id_type = if id_type.is_some() || input.ctx.is_none() {
+        id_type.unwrap()
+    } else {
+        &ctx_types
+    };
+    let deku_id = if deku_ids.is_empty() {
+        quote! {
+            impl #ident {
+                fn deku_id(&self) -> #deku_id_id_type {
+                    match self {
+                        _ => panic!("deku id for variant not found")
```

Comment:
We should avoid panics, a `DekuError` should be returned

Reply:
Fixed